### PR TITLE
Maybe fix crash because of #2309

### PR DESCRIPTION
**Pull request by TheNain38**
#2309

Could someone check if it fixes crash?

### CODE REVIEW

**Comment by tux3** [comment]
One second.
At first glance retranslateUi can be called even if Core isn't started, so you need to check that it's not a nullptr before calling getUsername/getStatusMessage on it

That would explain the crash. Did you try running it in a debugger?

**Comment by TheNain38** [comment]
No, I didn't try.

**Comment by TheNain38** [comment]
If it is `nullptr` it would set the username and status to empty "" ?

**Comment by tux3** [comment]
Alright, I ran it and it indeed crashes because tox is a nullptr.
I'll push a fix, but please try to be more careful next time!

**Comment by TheNain38** [comment]
@tux3 Yes, for sure, but I can't setup a build environement on my windows neither on my linux. Also I didn't had the time to see and understand all part of the code. Sorry, I will be more careful.
Also thanks for creating qTox. It's the best client ever. ;)

**Comment by tux3** [comment]
@TheNain38 okay, I understand. Thanks for contributing :)

**Comment by TheNain38** [comment]
@tux3 And also the `setUsername(core->getUsername());` and `setStatusMessage(core->getStatusMessage());` to avoid duplicated code?

**Comment by tux3** [comment]
@TheNain38 Yes, I pushed that too in 14235c5a19fbd3c8484ec16068ea33a72d64ff0a

**Comment by TheNain38** [comment]
@tux3 Ah, didn't see

**Comment by TheNain38** [comment]
@tux3 you can also delete the line 2035 in 'src/widget/widget.cpp'

**Comment by tux3** [comment]
@TheNain38 You're right, thanks!

**Comment by zetok** [comment]
@TheNain38 

> I can't setup a build environement on my windows neither on my linux.

If you have problem(s) with that, please say what kind of problem those are - that is, if you use distro that is sufficiently "new" to build qTox, instructions in INSTALL.md should be sufficient - if they're not, it's an issue and should be reported.

Also, if instructions are hard to follow, they should be improved, so it would be good if you could point out what is problematic `:)`

**Comment by TheNain38** [comment]
@zetok In my Windows 8.1 64-bits I have problems with missing filter_audio.h and vpx/vpx_image.h and some other I don't remember and on my linux mint I can't build toxcore.

**Comment by zetok** [comment]
Well, dunno about building on windows, but what is the problem with building toxcore on mint?

What **exactly** is the problem?

**Comment by TheNain38** [comment]
@zetok I will tell you more the next evening, I need to sleep and I'm on my phone, so I can't say much right now.
